elastic/kibana
# [Security Solution] [Platform] Utilize SO resolve api for reading rules by `id`

## Summary

Resolves https://github.com/elastic/kibana/issues/109191

------------------------------

### Testing `aliasMatch`

1. Start up kibana 7.14 and local Elasticsearch @ 7.14 which saves its data to a directory
2. Create a space in kibana
3. Create a detection rule (and create some alerts)
4. Copy the hyperlink for the rule details page somewhere
5. Shut down kibana 7.14 but DO NOT SHUTDOWN Elasticsearch
6. Start up this branch with the following key in your `kibana.dev.yml` -> `elasticsearch.ignoreVersionMismatch: true`
7. Log into the space you created the rule in
8. Visit the hyperlink you copied in step 4
9. You should see the aliasMatch toast appear.

<img width="1557" alt="alias_match_rule_details" src="https://user-images.githubusercontent.com/915763/135679569-58bcbc4f-7c46-483e-8ec3-479932bde596.png">


------------------------------

### Testing `conflict`

1. Create a rule using this branch
2. Go to https://github.com/elastic/kibana/pull/112478/files#diff-73b3c58b193075328ab707123d3e1ef1083df0260fc0881ac96d8c096d10244fR287 and change that line to be `const outcome = null; //rule.outcome`
3. Go to https://github.com/elastic/kibana/pull/112478/files#diff-73b3c58b193075328ab707123d3e1ef1083df0260fc0881ac96d8c096d10244fR306 and change that line to be `const outcome = 'conflict'; //rule?.outcome` This will mock a conflicting saved object _id
4. Kibana will restart, then visit the rule details page and you should see the below callout

conflict
<img width="1560" alt="redirect_callout" src="https://user-images.githubusercontent.com/915763/135679800-fdd46bfd-d78a-446d-9124-c02995ef342f.png">





### Checklist

Delete any items that are not applicable to this PR.



### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Rules with an aliasMatch will not have rule status saved objects associated with them until their next rule run | High | Medium | Update rule status code + routes to utilize SO resolve, migrate to using event log for statuses, or communicate to users that their statuses will not be reachable in 8.0 |


### For maintainers
